src/main/java/org/radarcns/management/web/rest/SourceTypeResource.java

```diff
@@ -224,14 +226,13 @@
         List<ProjectDTO> projects = sourceTypeService.findProjectsBySourceType(producer, model,
                 version);
         if (!projects.isEmpty()) {
-            throw new CustomParameterizedException(ErrorConstants.ERR_SOURCE_TYPE_IN_USE,
-                    projects.stream()
-                            .map(p -> p.getProjectName())
-                            .reduce((s1, s2) -> String.join(", ", s1, s2))
-                            .get()); // we know the list is not empty so calling get() is safe here
+            throw new InvalidRequestException(
+                // we know the list is not empty so calling get() is safe here
+                "Cannot delete a source-type that " + "is being used by project(s)", SOURCE_TYPE,
```

Comment:
Add back the list of projects here? The reduce(String.join).get() can be  replaced with collect(Collectors.joining)